src/components/ImageWithSizeCalculation.js

```diff
@@ -39,68 +44,29 @@ class ImageWithSizeCalculation extends PureComponent {
 
         this.imageLoadingStart = this.imageLoadingStart.bind(this);
         this.imageLoadingEnd = this.imageLoadingEnd.bind(this);
+        this.onError = this.onError.bind(this);
+        this.imageLoadedSuccessfuly = this.imageLoadedSuccessfuly.bind(this);
     }
 
-    componentDidMount() {
-        this.calculateImageSize();
-    }
-
-    componentDidUpdate(prevProps) {
-        if (prevProps.url === this.props.url) {
-            return;
-        }
-
-        this.calculateImageSize();
+    onError() {
+        Log.hmmm('Unable to fetch image to calculate size', {url: this.props.url});
     }
 
-    componentWillUnmount() {
-        if (!this.getImageSizePromise) {
-            return;
-        }
-
-        this.getImageSizePromise.cancel();
+    imageLoadingStart() {
+        this.setState({isLoading: true});
     }
 
-    /**
-     * @param {String} url
-     * @returns {Promise}
-     */
-    getImageSize(url) {
-        return new Promise((resolve, reject) => {
-            Image.getSize(url, (width, height) => {
-                resolve({width, height});
-            }, (error) => {
-                reject(error);
-            });
-        });
+    imageLoadingEnd() {
+        this.setState({isLoading: false});
     }
 
-    calculateImageSize() {
-        if (!this.props.url) {
+    imageLoadedSuccessfuly(event) {
+        if (!lodashGet(event, 'nativeEvent.width', false) || !lodashGet(event, 'nativeEvent.height', false)) {
+            // Image didn't load properly
             return;
         }
 
-        this.getImageSizePromise = makeCancellablePromise(this.getImageSize(this.props.url));
-        this.getImageSizePromise.promise
-            .then(({width, height}) => {
-                if (!width || !height) {
-                    // Image didn't load properly
-                    return;
-                }
-
-                this.props.onMeasure({width, height});
-            })
-            .catch((error) => {
-                Log.hmmm('Unable to fetch image to calculate size', {error, url: this.props.url});
-            });
-    }
-
-    imageLoadingStart() {
-        this.setState({isLoading: true});
-    }
-
-    imageLoadingEnd() {
-        this.setState({isLoading: false});
+        this.props.onMeasure({width: event.nativeEvent.width, height: event.nativeEvent.height});
```

Comment:
This check is contradicting:

1. `imageLoadSuccessfuly` is called from `onLoad` 
2. `onLoad` would be triggered only on success 

If that's not the case, we should address the problem in `onLoad` instead of doing a check here

Simply said - `onLoad` shouldn't be triggered when the image didn't load properly 

---

Seems this check is carried over from the original code
And it was introduced a long time ago here: 091abdc44b73bc9dc5880fbabac1855b8d925575

Like I said - if there's ever a case where `onLoad` (or `getSize`) was called without `width` and `height` - it is a bug in the underlying library and not something that should be handled in App 

_(For what I know for RN web - there won't be a case where `getSize` would not return `width` or `height`
@thomas-coldwell do you know whether `FastImage` needs this check?)_

@Beamanator @mountiny @tgolen what do you think guys?
- drop the check
- handle in a follow up
- do nothing / not big deal

---

I think this might be easier to understand:

```js
const {width, height} = lodashGet(event, 'nativeEvent' {});
if (!width || !height) {
// Image didn't load properly
return;

this.props.onMeasure({ width, height });
```

Or, if we don't want to destruct values
```js
const nativeEvent = lodashGet(event, 'nativeEvent' {});
if (!nativeEvent.width || !nativeEvent.height) {
...
```